### PR TITLE
Compare ignorecase endpointEncryption.Type

### PR DESCRIPTION
**Pull request by rigazilla**
This PR allows to uniform the EndpointEncryption.Type values to the CamelCase format

### CODE REVIEW

File: deploy/cr/tls/cr_minimal_with_tls.yaml [hunk 1]
```diff
@@ -7,10 +7,10 @@ spec:
   security:
     endpointEncryption:
             # Define the certificate to encrypt endpoint traffic.
-            # `type: service` uses platform service certificate.
-            # `type: secret` uses custom certificates.
-            type: secret
-            # If `type: service`, uncomment the following `certServiceName` line:
+            # `type: Service` uses platform service certificate.
+            # `type: Secret` uses custom certificates.
+            type: Secret
```
Comment (dmvolod) [inline deploy/cr/tls/cr_minimal_with_tls.yaml:12]: @rigazilla can we add validate for incorrect values, i.e. not Service and Secret in some place or it doesn't make sense?

Comment (rigazilla) [reply deploy/cr/tls/cr_minimal_with_tls.yaml:12]: @dmvolod  If it sounds good, I would plan a general assessment on yaml input validation for 2.1

Comment (dmvolod) [reply deploy/cr/tls/cr_minimal_with_tls.yaml:12]: Yes, sure, @rigazilla good plan.

**Comment by rigazilla** [comment]
merged, thanks for the review @dmvolod !

**Pull request merged**